### ISSUE TITLE
Problem with accept invite in calendar

### ISSUE THREAD

**Issue by hello-tty** [python-caldav/caldav#399]
Hello. I use a script to automatically accept calendar appointments, but recently it stopped working for me. Tell me what could be the problem?

Example code I'm using
```
import requests
import caldav
username = 'login'
password = 'password'

room_url = 'https://calendar_address/calendars'
client = caldav.DAVClient(
    url=room_url,
    auth=(username, password)
)

principal = client.principal()
calendars = principal.calendars()
my_calendar = principal.calendar(name='Common')

events = my_calendar.events()
for event in events:
    if 'test' in event.data:
        print(event.url)
        my_calendar.event_by_url(event.url).accept_invite(my_calendar)
```

The error I get
![image](https://github.com/python-caldav/caldav/assets/129526484/f784eea0-22b9-4f72-80de-d37ada90548f)

**Comment by tobixen** [comment]
The `accept_invite`-method is under-tested, I started working on this functionality and then it got deprioritized, sorry for that.

It seems like your calendar server does not want to yield your address.  Did this work earlier, and now it has stopped working, you said?

Is it by any chance possible for you to set up a testing account at your calendar server that I could use for test purposes?

**Comment by tobixen** [comment]
Sorry that things are taking time.

There were quite some of the tests failing towards your server, so I've created #401 and I'll look a bit into that before working with this.  (#400 is also relevant).

**Comment by tobixen** [comment]
I have reproduced the error in a test by now.

**Comment by tobixen** [comment]
Sorry for the slow progress here.   I do have a plan for how to solve this, but it seems like I'm overbooked for the rest of this week at least ...

**Comment by hello-tty** [comment]
Hello! Well, we'll wait. 
Do you have any idea how long it will take for the problem to be resolved?

**Comment by tobixen** [comment]
Sorry, been slightly overloaded with more important tasks recently.  When trying to look back into this, I get an authorization error from the server.  Is the test account still active?

**Comment by tobixen** [comment]
I move the "milestone" to the 3.0-release as there isn't much I can do until I'm able to reproduce the problem.  Will reschedule it to an earlier release if the problem pops up for me.  I do hope to have some time to allocate to the caldav project over the next few months.

**Comment by tobixen** [comment]
I won't do anything with it this year:

*  I need to put up a test environment with several accounts before I can do proper testing on features related to RFC 38
* I need either access to a server or a local test server where the problems here can be reproduced.

I asked Claude for an analysis on this issue, this is what it came up with:

# Issue #399 Analysis: Calendar Invite Acceptance Problem

**Date:** 2025-12-03
**Issue URL:** https://github.com/python-caldav/caldav/issues/399
**Status:** Open (Milestone 3.0)

## Issue Summary

This issue addresses a problem with the `accept_invite()` method failing when users attempt to automatically accept calendar appointments. The user reported that their script for automatically accepting calendar invitations stopped working recently.

---

## Is it still relevant?

**YES** - This is still relevant, but currently **blocked** due to lack of a reproducible test environment.

The maintainer (tobixen) has indicated they cannot proceed without access to a working test server that exhibits the problem. The original test account became inaccessible as of October 2024.

---

## Current State

### What exists:

1. ✅ **accept_invite() method** - Implemented in calendarobjectresource.py:609
2. ✅ **Related methods** - decline_invite() and tentatively_accept_invite() also exist
3. ✅ **Test coverage** - testInviteAndRespond() exists in test_caldav.py:604
4. ✅ **Example code** - scheduling_examples.py demonstrates usage (though not tested)
5. ✅ **Compatibility flags** - Several scheduling-related flags exist:
   - `no_scheduling` (RFC6638 not supported)
   - `no_scheduling_mailbox` (inbox/outbox not available)
   - `no_scheduling_calendar_user_address_set` (calendar user addresses not available)

### What's missing or problematic:

1. ❌ **No reproducible test case** - Cannot debug without access to failing server
2. ❌ **Under-tested functionality** - Maintainer explicitly stated accept_invite is "under-tested"
3. ❌ **calendar_user_address_set() issues** - The method attempts to get the user's calendar address, which may fail on some servers
4. ❌ **No caldav-server-tester checks** for scheduling features
5. ❌ **Unclear error conditions** - The error was reported as an image (content not visible), so exact failure mode is unknown
6. ❌ **Complex RFC6638 implementation** - Multiple TODO comments in calendarobjectresource.py:641-653 indicate uncertainty about correct implementation

---

## How accept_invite() Works

The implementation flow (calendarobjectresource.py:609-653):

1. Calls `_reply_to_invite_request("ACCEPTED", calendar)`
2. Verifies this is an invite request via `is_invite_request()`
3. Gets the calendar to save to (defaults to first calendar if not specified)
4. Pops the METHOD from icalendar_instance
5. Calls `change_attendee_status(partstat="ACCEPTED")`66

**Comment by tobixen** [comment]
Depends on #596 

I estimate it's about 8 hours work to clean up this properly.

**Comment by tobixen** [comment]
The invite handling has been hardened and better tested.  The test framework now includes docker containers for several multi-user open-source caldav server implementations:

```
tests/test_caldav.py::TestSchedulingForServerNextcloud::testInviteAndRespond PASSED                                                                     [  5%]
tests/test_caldav.py::TestSchedulingForServerNextcloud::testAcceptInviteUsernameEmailFallback SKIPPED (Attendee username 'user2' is not an email ad...) [ 10%]
tests/test_caldav.py::TestSchedulingForServerDavis::testInviteAndRespond PASSED                                                                         [ 15%]
tests/test_caldav.py::TestSchedulingForServerDavis::testAcceptInviteUsernameEmailFallback SKIPPED (Attendee username 'user2' is not an email addres...) [ 21%]
tests/test_caldav.py::TestSchedulingForServerZimbra::testInviteAndRespond PASSED                                                                        [ 26%]
tests/test_caldav.py::TestSchedulingForServerZimbra::testAcceptInviteUsernameEmailFallback PASSED                                                       [ 31%]
tests/test_caldav.py::TestSchedulingForServerCCS::testInviteAndRespond PASSED                                                                           [ 36%]
tests/test_caldav.py::TestSchedulingForServerCCS::testAcceptInviteUsernameEmailFallback SKIPPED (Attendee username 'user02' is not an email address...) [ 42%]
tests/test_caldav.py::TestSchedulingForServerStalwart::testInviteAndRespond PASSED                                                                      [ 47%]
tests/test_caldav.py::TestSchedulingForServerStalwart::testAcceptInviteUsernameEmailFallback SKIPPED (Attendee username 'user2' is not an email add...) [ 52%]
tests/test_caldav.py::TestSchedulingForServerDavical::testInviteAndRespond PASSED                                                                       [ 57%]
tests/test_caldav.py::TestSchedulingForServerDavical::testAcceptInviteUsernameEmailFallback SKIPPED (Attendee username 'user2' is not an email addr...) [ 63%]
tests/test_caldav.py::TestSchedulingForServerSOGo::testInviteAndRespond PASSED                                                                          [ 68%]
tests/test_caldav.py::TestSchedulingForServerSOGo::testAcceptInviteUsernameEmailFallback SKIPPED (Attendee username 'user2' is not an email address...) [ 73%]
tests/test_caldav.py::TestSchedulingForServerCyrus::testInviteAndRespond PASSED                                                                         [ 78%]
tests/test_caldav.py::TestSchedulingForServerCyrus::testAcceptInviteUsernameEmailFallback SKIPPED (Attendee username 'user2' is not an email addres...) [ 84%]
tests/test_caldav.py::TestSchedulingForServerBaikal::testInviteAndRespond PASSED                                                                        [ 89%]
tests/test_caldav.py::TestSchedulingForServerBaikal::testAcceptInviteUsernameEmailFallback SKIPPED (Attendee username 'user2' is not an email addre...) [ 94%]
tests/test_caldav_unit.py::TestAsyncCalendarObjectResource::test_accept_invite_raises_not_implemented_for_async_client PASSED                           [100%]
```

I assume accepting invites should work now, so I'll close this issue.
